datalad_metalad/add.py

```diff
@@ -500,7 +504,7 @@ def add_finite_set(metadata_objects: List[JSONType],
                 continue
 
             if len(result) == 1:
-                yield result[0]
+                yield get_status_dict(**result[0])
```

Comment:
Since no keyword arguments are provided, ``get_status_dict(**result[0]))`` would just return a copy of ``result[0]``. Therefore I would leave it at ``yield result[0]``.